lib/pipeline.js

```diff
@@ -139,15 +146,24 @@ class PipelineModel extends BaseModel {
 
         const factory = JobFactory.getInstance();
 
-        const jobsToCreate = jobList.map(j =>
-            this.getConfiguration(j.ref).then((config) => {
-                const jobConfig = {
-                    pipelineId: this.id,
-                    name: j.name,
-                    permutations: config.jobs.main
-                };
+        const jobsToCreate = prList.map(pr =>
+            this.getConfiguration(pr.ref).then((config) => {
+                const prJobNames = workflowParser.getNextJobs(config.workflowGraph, {
+                    trigger: '~pr',
+                    prNum: pr.name.split('-')[1]
+                });
+
+                return Promise.all[prJobNames.map((name) => {
+                    const jobName = name.match(PR_JOB_NAME)[PR_NAME_GROUP.job];
```

Comment:
abstract this behind a function `getNameFromJob(job, type)`  as I've described in previous comment.